pkg/controllers/management/feature/feature_handler.go

```diff
@@ -0,0 +1,100 @@
+package feature
+
+import (
+	"context"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/rancher/pkg/features"
+	managementv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type handler struct {
+	featuresClient managementv3.FeatureClient
+	tokensLister   managementv3.TokenCache
+	tokenEnqueue   func(string, time.Duration)
+}
+
+func Register(ctx context.Context, wContext *wrangler.Context) {
+	h := handler{
+		featuresClient: wContext.Mgmt.Feature(),
+		tokensLister:   wContext.Mgmt.Token().Cache(),
+		tokenEnqueue:   wContext.Mgmt.Token().EnqueueAfter,
+	}
+	wContext.Mgmt.Feature().OnChange(ctx, "feature-handler", h.sync)
+}
+
+func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	obj, err := h.setLockedValue(obj)
+	if err != nil {
+		return obj, err
+	}
+
+	if obj.Name == features.TokenHashing.Name() {
+		return obj, h.refreshTokens()
+	}
+	return obj, nil
+}
+
+func (h *handler) refreshTokens() error {
+	tokenList, err := h.tokensLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, token := range tokenList {
+		if token.Labels != nil && token.Labels[tokens.TokenHashed] == "true" {
+			continue
+		}
+		h.tokenEnqueue(token.Name, 10*time.Second)
+	}
+	return nil
+}
+
+// setLockedValue evaluates whether a value should be written to the lockedValue
+// field on status and records the value if so.
+func (h *handler) setLockedValue(obj *v3.Feature) (*v3.Feature, error) {
+	lockedValue := LockedValue(obj)
+
+	if lockedValue == nil && obj.Status.LockedValue == nil {
+		return obj, nil
+	}
+	update := (lockedValue == nil && obj.Status.LockedValue != nil) || (lockedValue != nil && obj.Status.LockedValue == nil) ||
```

Comment:
Can you comment the cases here, I can't understand the first case. If lockedValue returns nil the feature can be changed (according to that method) so how would you ever get here. The last case is also confusing me.

Reply:
There is no telling what locked logic other feature flags may use; you can't change spec if there's a locked value due to validator but you could change something else. This is just making sure that LockedValue on status matches the value returned from the function.

Reply:
So this makes sense to me: 
- Validator is simply to validate if status.LockedValue is not nil, we allow spec.Value to change only if it's nil.
- setLockedValue is actually setting obj.StatusLockedValue from spec.Value, and we we keep them in sync.  

But I feel like I don't fully grasp what we're allowing features to do with LockedValue. Is the intention that they can introduce and some other field on spec....but on status, it won't be a corresponding new field and it'll reflect on LockedValue? In that case, is LockedValue really locked then?

Reply:
Another reason for this logic is upgrade, it's possible the LockedValue function is updated in a new version and the LockedValue will need to be updated.

Reply:
I'm also confused at the real goal of all of this. It seems to me that its all around pinning a spec once a certain value is hit. But the real problem we are trying to solve is: users need to know token-hashing is a one-way feature and they shouldn't be able to break that.
If we can make it so it doesn't matter if they flip the switch back and forth, and "one way" is in the description that would also satisfy the problem. Meaning: On disable this feature we log an error. On enabling this feature if a v3 token has the `hashFormat` we know its already on and we can't redo it and we also log an error. Would UX for that be any different than this ? 
We can go this route if we need to but it seems complicated.

Reply:
The user is already prevented from toggling the value in the UI due to the validator. This is just making it so if for some reason the LockedValue should be different it updates it accordingly. This could happen due to an upgrade.

Reply:
Talked offline and the real reason is so we can flip the code on/off based on the feature flag, rather than based on token string.